src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -84,31 +80,41 @@ public SecurityIndexAccessEvaluator(final Settings settings, AuditLog auditLog,
         securityIndexDeniedActionPatternsListNoSnapshot.add("indices:admin/close*");
         securityIndexDeniedActionPatternsListNoSnapshot.add("cluster:admin/snapshot/restore*");
 
-        securityDeniedActionMatcher = WildcardMatcher.from(restoreSecurityIndexEnabled ? securityIndexDeniedActionPatternsList : securityIndexDeniedActionPatternsListNoSnapshot);
+        securityDeniedActionMatcher = WildcardMatcher.from(
+                restoreSecurityIndexEnabled ? securityIndexDeniedActionPatternsList : securityIndexDeniedActionPatternsListNoSnapshot);
     }
-    
+
     public PrivilegesEvaluatorResponse evaluate(final ActionRequest request, final Task task, final String action, final Resolved requestedResolved,
-            final PrivilegesEvaluatorResponse presponse)  {
-        final boolean isDebugEnabled = log.isDebugEnabled();
+        final PrivilegesEvaluatorResponse presponse,  ConfigModelV7.SecurityRoles securityRoles) {
+
+            final boolean isDebugEnabled = log.isDebugEnabled();
+
+            if( matchAnySystemIndices(requestedResolved) && !checkSystemIndexPermissionsForUser(securityRoles)){
+            log.warn("An account without the {} permission  is trying to access a System Index. Related indexes: {}", ConfigConstants.SYSTEM_INDEX_PERMISSION, requestedResolved.getAllIndices() );
```

Comment:
Is it possible to filter out the actual SystemIndices here and just log those? Unless I am mistaken, `requestedResolved.getAllIndices()` will return all indices touched by the request. It may be more apporpraite to only return the System Indices since you state "Related Indices: ..."

Alternatively, swap to "Requested indices: ... " ?